wordpress-mobile/WordPress-Android
# Fixes: User not guided to Proper tab on Navigation

Fixes #16288 

This PR fixes the issue where there is no indication, for certain tasks, that the user needs to switch tabs to see the focus point and complete the task. 

This PR introduces a quick start focus point on the Tab header and Snackbar notice with the message to the User to switch to the Tab which holds the Quick start focus point for the Task. 

In a few tasks, if the user changes tabs during the completion of the tasks (it could be either intentionally or accidentally) we are not indicating that the current focus point is in the other tab. The issue will only happen when a tab different from the Initial Screen is selected before the user clicks GO to start the affected tasks. 

The below video capture shows the UI behaviour after the fix. The tab in which the Quickstart focus point is shown is switched and the GO button is clicked. A quick start focus point on SIte Menu tab/ Home tab to indicate the user to switch to the tab. 

https://user-images.githubusercontent.com/17463767/166743251-77c2058a-63c2-4a29-aa0a-e95cc4162b65.mp4

To test:
Steps to reproduce the behavior
Initial Screen: Site Menu
- Complete Customize your site > Check your site title.
- Tap GO to start the next task.
- Complete Customize your site > Choose a unique site icon.
- Tap the HOME tab.
- Tap GO to start the next task.
- Notice that focus point is displayed.
- Tap SITE MENU tab.
- Notice that the focus point is displayed over Pages quick link.

Initial Screen: HOME
- Complete Customize your site > Check your site title.
- Tap GO to start the next task.
- Complete Customize your site > Choose a unique site icon.
- Tap the SITE MENU tab.
- Tap GO to start the next task.
- Notice that the focus point is displayed.
- Tap the HOME tab.
- Notice that the focus point is displayed over the Pages quick link ribbon card.

Merging Instructions:
The issue was introduced as part of the changes for the 19.8 release. It'd be good if we can target frozen branch 19.8.
/cc @AliSoftware


## Regression Notes
1. Potential unintended areas of impact
Quick start focus point not being shown properly 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests 

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
